src/instantsearch/widget.js

```diff
@@ -10,34 +10,104 @@ class AlgoliaPlacesWidget {
     }
 
     this.placesOptions = placesOptions;
+    this.placesAutocomplete = places(this.placesOptions);
+
+    this.state = {};
   }
-  init({ helper }) {
-    const placesAutocomplete = places(this.placesOptions);
 
+  init({ helper }) {
+    console.log('Hello:init');
     helper
       .setQueryParameter('insideBoundingBox')
-      .setQueryParameter('aroundLatLng', this.defaultPosition);
+      .setQueryParameter(
+        'aroundLatLng',
+        this.state.position || this.defaultPosition
+      );
 
-    placesAutocomplete.on('change', opts => {
+    this.placesAutocomplete.on('change', opts => {
       const {
         suggestion: {
           latlng: { lat, lng },
+          value,
         },
       } = opts;
 
+      this.state.position = `${lat},${lng}`;
+      this.state.query = value;
+
+      console.log('Hello:change');
       helper
         .setQueryParameter('insideBoundingBox')
-        .setQueryParameter('aroundLatLng', `${lat},${lng}`)
+        .setQueryParameter('aroundLatLng', this.state.position)
         .search();
     });
 
-    placesAutocomplete.on('clear', () => {
+    this.placesAutocomplete.on('clear', () => {
+      this.state.position = undefined;
+      this.state.query = undefined;
+
+      console.log('Hello:clear');
       helper
         .setQueryParameter('insideBoundingBox')
         .setQueryParameter('aroundLatLng', this.defaultPosition)
         .search();
     });
   }
+
+  getWidgetSearchParameters(searchParameters, { uiState }) {
+    console.log('gWSP', 'current', this.state, 'future', uiState.places);
+    if (
+      !uiState.places ||
+      (uiState.places.position === this.state.position &&
+        uiState.places.query === this.state.query)
+    ) {
+      this.placesAutocomplete.setVal('');
```

Comment:
The condition is when uistate.places is not define or position and query are the same which doesn't mean that the query is empty.